File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/ChatApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class ChatApi {
    private final HttpClient client;
    
    public ChatApi(HttpClient client) {
        this.client = client;
    }

    /** Get chat completion */
    public ChatCompletionResponse getCompletion(String completion_id) throws Exception {
        return (ChatCompletionResponse) client.get(ApiPaths.aiPath("/chat/completions/" + completion_id + ""));
    }

    /** Update chat completion */
    public ChatCompletionResponse createUpdateCompletion(String completion_id, UpdateCompletionPostRequest body) throws Exception {
        return (ChatCompletionResponse) client.post(ApiPaths.aiPath("/chat/completions/" + completion_id + ""), body);
    }

    /** Delete chat completion */
    public ChatCompletionDeleteResponse deleteCompletion(String completion_id) throws Exception {
        return (ChatCompletionDeleteResponse) client.delete(ApiPaths.aiPath("/chat/completions/" + completion_id + ""));
    }

    /** Update chat completion */
    public ChatCompletionResponse patchUpdateCompletion(String completion_id, UpdateCompletionPatchRequest body) throws Exception {
        return (ChatCompletionResponse) client.patch(ApiPaths.aiPath("/v1/chat/completions/" + completion_id + ""), body);
    }

    /** Update chat completion */
    public ChatCompletionResponse patchUpdateCompletionChat(String completion_id, PatchUpdateCompletionRequest body) throws Exception {
        return (ChatCompletionResponse) client.patch(ApiPaths.aiPath("/chat/completions/" + completion_id + ""), body);
    }

    /** Get chat completion */
    public ChatCompletionResponse getChatCompletion(String completion_id) throws Exception {
        return (ChatCompletionResponse) client.get(ApiPaths.aiPath("/management/chat/completions/" + completion_id + ""));
    }

    /** Update chat completion */
    public ChatCompletionResponse updateChatCompletion(String completion_id, UpdateChatCompletionRequest body) throws Exception {
        return (ChatCompletionResponse) client.post(ApiPaths.aiPath("/management/chat/completions/" + completion_id + ""), body);
    }

    /** Delete chat completion */
    public ChatCompletionDeleteResponse deleteChatCompletion(String completion_id) throws Exception {
        return (ChatCompletionDeleteResponse) client.delete(ApiPaths.aiPath("/management/chat/completions/" + completion_id + ""));
    }

    /** List chat completions */
    public ChatCompletionList listCompletions(Map<String, Object> params) throws Exception {
        return (ChatCompletionList) client.get(ApiPaths.aiPath("/chat/completions"), params);
    }

    /** Create chat completion */
    public CreateChatCompletionResponse createChatCompletion(ChatCompletionRequest body) throws Exception {
        return (CreateChatCompletionResponse) client.post(ApiPaths.aiPath("/chat/completions"), body);
    }

    /** Get chat messages */
    public ChatMessageList getChatMessages(String completion_id, Map<String, Object> params) throws Exception {
        return (ChatMessageList) client.get(ApiPaths.aiPath("/management/chat/completions/" + completion_id + "/messages"), params);
    }

    /** List chat completions */
    public ChatCompletionList listChatCompletions(Map<String, Object> params) throws Exception {
        return (ChatCompletionList) client.get(ApiPaths.aiPath("/management/chat/completions"), params);
    }

    /** Get chat completion messages */
    public ChatMessageList getMessages(String completion_id, Map<String, Object> params) throws Exception {
        return (ChatMessageList) client.get(ApiPaths.aiPath("/chat/completions/" + completion_id + "/messages"), params);
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/ChatApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class ChatApi(private val client: HttpClient) {

    /** Get chat completion */
    suspend fun getCompletion(completion_id: String): ChatCompletionResponse? {
        return client.get(ApiPaths.aiPath("/chat/completions/$completion_id")) as? ChatCompletionResponse
    }

    /** Update chat completion */
    suspend fun createUpdateCompletion(completion_id: String, body: UpdateCompletionPostRequest): ChatCompletionResponse? {
        return client.post(ApiPaths.aiPath("/chat/completions/$completion_id"), body) as? ChatCompletionResponse
    }

    /** Delete chat completion */
    suspend fun deleteCompletion(completion_id: String): ChatCompletionDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/chat/completions/$completion_id")) as? ChatCompletionDeleteResponse
    }

    /** Update chat completion */
    suspend fun patchUpdateCompletion(completion_id: String, body: UpdateCompletionPatchRequest): ChatCompletionResponse? {
        return client.patch(ApiPaths.aiPath("/v1/chat/completions/$completion_id"), body) as? ChatCompletionResponse
    }

    /** Update chat completion */
    suspend fun patchUpdateCompletionChat(completion_id: String, body: PatchUpdateCompletionRequest): ChatCompletionResponse? {
        return client.patch(ApiPaths.aiPath("/chat/completions/$completion_id"), body) as? ChatCompletionResponse
    }

    /** Get chat completion */
    suspend fun getChatCompletion(completion_id: String): ChatCompletionResponse? {
        return client.get(ApiPaths.aiPath("/management/chat/completions/$completion_id")) as? ChatCompletionResponse
    }

    /** Update chat completion */
    suspend fun updateChatCompletion(completion_id: String, body: UpdateChatCompletionRequest): ChatCompletionResponse? {
        return client.post(ApiPaths.aiPath("/management/chat/completions/$completion_id"), body) as? ChatCompletionResponse
    }

    /** Delete chat completion */
    suspend fun deleteChatCompletion(completion_id: String): ChatCompletionDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/management/chat/completions/$completion_id")) as? ChatCompletionDeleteResponse
    }

    /** List chat completions */
    suspend fun listCompletions(params: Map<String, Any>? = null): ChatCompletionList? {
        return client.get(ApiPaths.aiPath("/chat/completions"), params) as? ChatCompletionList
    }

    /** Create chat completion */
    suspend fun createChatCompletion(body: ChatCompletionRequest): CreateChatCompletionResponse? {
        return client.post(ApiPaths.aiPath("/chat/completions"), body) as? CreateChatCompletionResponse
    }

    /** Get chat messages */
    suspend fun getChatMessages(completion_id: String, params: Map<String, Any>? = null): ChatMessageList? {
        return client.get(ApiPaths.aiPath("/management/chat/completions/$completion_id/messages"), params) as? ChatMessageList
    }

    /** List chat completions */
    suspend fun listChatCompletions(params: Map<String, Any>? = null): ChatCompletionList? {
        return client.get(ApiPaths.aiPath("/management/chat/completions"), params) as? ChatCompletionList
    }

    /** Get chat completion messages */
    suspend fun getMessages(completion_id: String, params: Map<String, Any>? = null): ChatMessageList? {
        return client.get(ApiPaths.aiPath("/chat/completions/$completion_id/messages"), params) as? ChatMessageList
    }
}
